docs/specification.md

```diff
@@ -0,0 +1,1199 @@
+        (or %tokens
+          (unit %allTokens)
+          (set %someTokens nat)
+        )
+    ))
+  )
+)
+```
+
+- Add or Remove token operators for the specified owners and token
+  types.
+
+- If two different commands in the list add and remove an operator
+  for the same token type, then the last one must take effect.
+
+- It's possible to update an operator for some specific tokens, or
+  to all tokens (TODO: discuss).
+
+- Sender and operation receiver must be whitelisted.
+
+- Contract must not be paused.
+
+**approveOperatorUpdate (TODO)**
+
+Types
+```
+tokenId
+
+operatorTokens =
+  | AllTokens
+  | SomeTokens (set tokenId)
+
+operatorParam =
+  ( address        :owner
+  , address        :operator
+  , operatorTokens :tokens
+  )
+
+approveOperatorUpdate = operatorParam
+```
+
+Parameter (in Michelson):
+```
+(pair %approveOperatorUpdate
+  (pair %operator
+      (address %owner)
+      (pair
+        (address %operator)
+        (or %tokens
+          (unit %allTokens)
+          (set %someTokens nat)
+        )
+    ))
+  (contract %callback
+    (pair %operator
+      (address %owner)
+      (pair
+        (address %operator)
+        (or %tokens
+          (unit %allTokens)
+          (set %someTokens nat)
+        )
+      )
+    )
+  )
+)
+```
+
+- Enables `operator` to operate on tokens held by `owner`.
+
+- `owner` must be whitelisted.
+
+- Contract must not be paused.
+
+**isOperator**
+
+Types
+```
+tokenId = nat
+
+operatorTokens =
+  | AllTokens
+  | SomeTokens (set tokenId)
+
+operatorParam =
+  ( address        :owner
+  , address        :operator
+  , operatorTokens :tokens
+  )
+
+isOperatorResponse =
+  ( operatorParam :operator
+  , bool          :isOperator
+  )
+
+isOperatorParam =
+  ( operatorParam               :operator
+  , contract isOperatorResponse :callback
+  )
+
+isOperator = isOperatorParam
+```
+
+Parameter (in Michelson):
+```
+(pair %isOperator
+  (pair %operator
+    (address %owner)
+    (pair
+      (address %operator)
+      (or %tokens
+        (unit %allTokens)
+        (set %someTokens nat)
+      )
+  ))
+  (contract %callback
+    (pair
+      (pair %operator
+        (address %owner)
+        (pair
+          (address %operator)
+          (or %tokens
+            (unit %allTokens)
+            (set %someTokens nat)
+          )
+      ))
+      (bool %isOperator)
+    )
+  )
+)
+```
+
+- Inspect if an address is an operator for the specified owner and
+  token types.
+
+- If the adddress is not an operator for at least one
+  requested token type, then the result is `false`.
+
+- It's possible to make this query for some specific tokens, or to
+  all tokens.
+
+**pause**
+
+Types
+```
+pause = (address :sender)
+```
+
+Parameter (in Michelson)
+```
+(pair %pause
+  (address %sender)
+  unit
+)
+```
+
+- Pauses transferring, burning and minting operations so
+  that they cannot be performed. All other operations remain
+  unaffected.
+
+- Cannot be called multiple times.
```

Comment:
I guess you mean that it can't be called multiple times without `unpause` call between `pause` calls. But can be called multiple times during contract's lifetime.